### PR TITLE
Replace groovy.json.JsonOutput with kotlinx.serialization.json in JsonReporter #74

### PR DESCRIPTION
**Pull request by jglaszka**
related to https://github.com/platan/tests-execution-chart/issues/74

### CODE REVIEW

File: tests-execution-chart-commons/src/test/groovy/io/github/platan/tests_execution_chart/reporters/json/JsonReporterTest.groovy [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import io.github.platan.tests_execution_chart.report.data.TimedTestResult
+import spock.lang.Specification
+import java.time.Instant
+
+import static io.github.platan.tests_execution_chart.report.data.TimedTestResult.Type.TEST
+
+class JsonReporterTest extends Specification {
+    def "should generate report in json"() {
+        given:
+        def PATHNAME = "./test"
```
Comment (3750) [inline tests-execution-chart-commons/src/test/groovy/io/github/platan/tests_execution_chart/reporters/json/JsonReporterTest.groovy:14]: Can we make `PATHNAME` and `CONFIG_OUTPUT_LOCATION` lowercase?

File: tests-execution-chart-commons/src/test/groovy/io/github/platan/tests_execution_chart/reporters/json/JsonReporterTest.groovy [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            it.setConfiguration(new JsonConfig(true, CONFIG_OUTPUT_LOCATION))
+        }
+
+        when:
+        reporter.report(report, new File(PATHNAME), "taskname")
+
+        then:
+        def file = new File("""${PATHNAME}/${CONFIG_OUTPUT_LOCATION}/taskname.json""")
+        // i dont know how to make it with indentation
+        file.getText("UTF-8") ==
```
Comment (3750) [inline tests-execution-chart-commons/src/test/groovy/io/github/platan/tests_execution_chart/reporters/json/JsonReporterTest.groovy:29]: `stripMargin` to the rescue https://blog.mrhaki.com/2010/06/groovy-goodness-strip-leading-spaces_20.html

```groovy
        file.getText("UTF-8") ==
                """{
                |    "results": [
                |        {
                |            "className": "class",
                |            "testName": "test",
                |            "startTime": 1678474802000,
                |            "endTime": 1678474805000,
                |            "resultType": "passed",
                |            "type": "TEST"
                |        }
                |    ],
                |    "marks": [
                |        {
                |            "name": "mark1",
                |            "timestamp": 1678474805000
                |        }
                |    ]
                |}""".stripMargin()
```

**Review by 3750 — CHANGES_REQUESTED** [review]
Looks really nice. Small suggestions regarding test.

File: tests-execution-chart-commons/src/test/groovy/io/github/platan/tests_execution_chart/reporters/json/JsonReporterTest.groovy (hunk 1 above)
Comment (jglaszka) [reply tests-execution-chart-commons/src/test/groovy/io/github/platan/tests_execution_chart/reporters/json/JsonReporterTest.groovy:14]: yes

File: tests-execution-chart-commons/src/test/groovy/io/github/platan/tests_execution_chart/reporters/json/JsonReporterTest.groovy (hunk 2 above)
Comment (jglaszka) [reply tests-execution-chart-commons/src/test/groovy/io/github/platan/tests_execution_chart/reporters/json/JsonReporterTest.groovy:28]: thanks, done

**Review by platan — APPROVED** [review]

**Comment by platan** [comment]
@jglaszka thank you for your contribution.

**Pull request merged**